File: dayThree/python/distinct_characters_count.py


def count_distinct_in(word) :
    lower_word = word.lower()
    total = 0
    alpha = "abcdefghijklmnopqrstuvwxyz0123456789"
    for index in alpha:
        character = index
        duplicate_counter = 0
        
        for second_index in lower_word:
            if character == second_index:
                duplicate_counter += 1
        if duplicate_counter > 1 :
            total += 1
    return total

word = "a11BAcb"

print(count_distinct_in(word))  

File: dayThree/java/DistinctCharactersCount.java
public class DistinctCharactersCount {

    public static void main(String[] args) {
        String word = "a111BAcb";
        System.out.println(countDistinct(word));
    }
    public static int countDistinct(String  word) {
        String lowerWord = word.toLowerCase();
        int total = 0;
        String alnum = "abcdefghijklmnopqrstuvwxyz0123456789";
        int duplicateCounter = 0; 
        for(int index = 0; index < alnum.length(); index++) {
            char character = alnum.charAt(index);
            duplicateCounter = 0;
            for(int secondIndex = 0; secondIndex < word.length(); secondIndex++){
                if(character == lowerWord.charAt(secondIndex)) {
                    duplicateCounter++;
                }
            }
            
            if(duplicateCounter > 1) {
                total++;
            }
            
        }
        return total;
    }
    
}
